src/formatting/string.rs

```diff
@@ -225,7 +225,25 @@ fn not_whitespace_except_line_feed(g: &str) -> bool {
 /// character is either a punctuation or a whitespace.
 /// FIXME(issue#3281): We must follow UAX#14 algorithm instead of this.
 fn break_string(max_width: usize, trim_end: bool, line_end: &str, input: &[&str]) -> SnippetState {
-    let break_at = |index /* grapheme at index is included */| {
+    let break_at = |index_in /* grapheme at index is included */| {
+        // Ensure break is not after an escape '\' as it will "escape" the `\` that is added
+        // for concatenating the two parts of the broken line.
+        let index = if (input[index_in] as &str).ne("\\") {
+            index_in
+        } else {
+            let index_offset = match input[0..index_in]
+                .iter()
+                .rposition(|grapheme| grapheme.ne(&"\\"))
+            {
+                // There is a non-`\` to the left
+                Some(non_backslash_index) => (index_in - non_backslash_index) % 2,
+                // Only `\` to the left
+                None => index_in % 2,
+            };
+            // Make sure break is after even number (including zero) of `\`
+            index_in - index_offset
+        };
```

Comment:
Admittedly another nit, but I think keeping `index` as the closure param name and just shadowing it with the assignment on 231 would be fine, especially since it's a usize

Reply:
OK. submitted the change.